src/controllers/createUser.ts

```diff
@@ -0,0 +1,28 @@
+import { IncomingMessage, ServerResponse } from 'http';
+import createNew from '../models/createNew';
+
+const createUser = async (req: IncomingMessage, res: ServerResponse) => {
+  try {
+    let body = '';
+    req.on('data', (chunk) => {
+      body += chunk.toString();
+    });
+    req.on('end', async () => {
+      const { username, age, hobbies } = JSON.parse(body);
+
+      const user = {
+        username,
+        age,
+        hobbies,
+      };
```

Comment:
here you have to make sure that username, age and hobbies are not undefined, and if they are send a response with 404